source/particle/world.cc

```diff
@@ -501,6 +501,53 @@ namespace aspect
     }
 
     template <int dim>
+    bool
+    World<dim>::particle_is_in_cell(const Particle<dim> &particle,
+                                    const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell) const
+    {
+      try
+        {
+          const Point<dim> p_unit = this->get_mapping().transform_real_to_unit_cell(cell, particle.get_location());
+          if (GeometryInfo<dim>::is_inside_unit_cell(p_unit))
+            return true;
+        }
+      catch (...)
+        {}
+      return false;
+    }
+
+    template <int dim>
+    std::map<double, typename parallel::distributed::Triangulation<dim>::active_cell_iterator>
+    World<dim>::neighbor_cells_to_search(const Particle<dim> &particle,
+                                         const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell) const
+    {
+      std::map<double,typename parallel::distributed::Triangulation<dim>::active_cell_iterator> neighbor_cells;
```

Comment:
(This will happen rarely and we still get the same result, but we may as well treat it right. )